Fix type hint syntax unsupported in Python 3.9

**Describe the bug**
A bug was recently introduced in our Db2 support. It slipped through the net because we can no longer test on Python 3.9.

The error:
```
third_party/ibis/ibis_db2/__init__.py:193: in Backend
    database: str | None = None,
E   TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

**What version of DVT are you using?**
8.5.1

**What type of connections are you using for source and target?**
Db2

--- a/tests/system/data_sources/test_db2.py
+++ b/tests/system/data_sources/test_db2.py
@@ -41,15 +41,18 @@ pytestmark = pytest.mark.flaky(
 )
 
 DB2_HOST = os.getenv("DB2_HOST", "localhost")
+DB2_USER = os.getenv("DB2_USER", "db2inst1")
 DB2_PASSWORD = os.getenv("DB2_PASSWORD")
+DB2_DATABASE = os.getenv("DB2_DATABASE", "testdb")
+DB2_PORT = os.getenv("DB2_PORT", 50000)
 
 CONN = {
     consts.SOURCE_TYPE: consts.SOURCE_TYPE_DB2,
     "host": DB2_HOST,
-    "user": "db2inst1",
+    "user": DB2_USER,
     "password": DB2_PASSWORD,
-    "port": 50000,
-    "database": "testdb",
+    "port": DB2_PORT,
+    "database": DB2_DATABASE,
 }
 
 

